### PR TITLE
Add interactive power analysis using Bokeh

### PR DESCRIPTION
**Pull request by gibsramen**
Adds code to generate a Bokeh app with Python backend.

### CODE REVIEW

**Comment by gibsramen** [comment]
So looks like `shutil.copytree` only added the `dirs_exist_ok` argument in 3.8. I think we can remove this argument and just make sure the user knows that intermediate directories won't be created.

**Comment by gibsramen** [comment]
@ahdilmore Could you look over the Bokeh code and also try creating an app with some of your own data?

File: evident/diversity_handler.py [hunk 1]
```diff
@@ -327,7 +327,7 @@ def __init__(
 
     def subset_values(self, ids: list) -> np.array:
         """Get alpha-diversity differences among provided samples."""
-        return self.data.loc[ids].values
+        return np.squeeze(self.data.loc[ids].values)
```
Comment (ahdilmore) [inline evident/diversity_handler.py:330]: Can you elaborate on why this change was necessary? not clear to me why you would need np.squeeze here

File: evident/interactive.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    shutil.copytree(support_files, output)
+    data_dir = os.path.join(output, "data")
+    os.mkdir(data_dir)
+
+    md = diversity_handler.metadata
+    # Get all valid categorical columns and save metadata to data dir
+    valid_cols = [
+        x for x in md.columns
+        if md[x].dtype == np.dtype("object")
+        and 1 < len(md[x].dropna().unique()) <= max_levels_per_category
```
Comment (ahdilmore) [inline evident/interactive.py:42]: Might be nice here to output warnings or some other type of notice that specific columns won't be included because they have more than 5 levels. 

Also is there a specific reason 5 was chosen? Besides that this is probably good for visualization purposes?

File: README.md [hunk 3]
```diff
@@ -139,6 +139,32 @@ When we inspect this plot, we can see how many samples we would need to collect
...
+```
+
+This will save the necessary files into a new directory `app/`.
+Navigate to the directory containing `app/` (**not** `app/` itself) and execute this command from your terminal:
+
+```bash
+bokeh serve --show app
+```
+
+This should open up a browser window where you can modify the chosen column, significance, level, and observations.
```
Comment (ahdilmore) [inline README.md:160]: Might also be helpful to clarify evident's behavior w.r.t max_levels_per_category somewhere in the README (since it has a default value this is something I can see myself & others overlooking!)

**Review by ahdilmore — COMMENTED** [review]
@gibsramen Made a few small comments/questions, but overall looks good! Will try this on some of my data.

File: evident/interactive.py (hunk 2 above)
Comment (ahdilmore) [reply evident/interactive.py:42]: Also - might be good to also check the number of unique values in each level? I can imagine that these plots would look strange if levels are severely imbalanced or if some levels had only a few entries.

File: evident/diversity_handler.py (hunk 1 above)
Comment (gibsramen) [reply evident/diversity_handler.py:330]: So this was because of a weird bug that I didn't actually understand. I looked into it and I think I know what happened. When saving the alpha diversity data to `diversity.alpha.tsv`, upon loading this becomes a `DataFrame` rather than a `Series`. As a result, the subsetting returns a `(N, 1)` array rather than `(N,)`. The `squeeze` was coercing this into the proper format but I think what we *should* do is check to make sure that the data passed into `AlphaDiversityHandler` is a `Series` rather than a `DataFrame`.

Thanks for pointing this out! This could have been a pain later down the line.

File: evident/__init__.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 from .diversity_handler import AlphaDiversityHandler, BetaDiversityHandler
 
 
-__version__ = "0.1.1"
+__version__ = "0.1.2dev"
```
Comment (ahdilmore) [inline evident/__init__.py:5]: Note that I had some issues with pandas 0.25.3

Notably received following error: 
FutureWarning: The signature of `[Series.to](http://series.to/)_csv` was aligned to that of `[DataFrame.to](http://dataframe.to/)_csv`, and argument ‘header’ will change its default value from False to True: please pass an explicit value to suppress this warning.
 [data.to](http://data.to/)_csv(data_loc, sep=“\t”, index=True)

And interactive plots showed up as blank. 

Issue resolved when updated to pandas 1.3.5. Require updated pandas?

File: evident/interactive.py (hunk 2 above)
Comment (gibsramen) [reply evident/interactive.py:42]: Yeah I figured that often we have columns with like dozens or hundreds of levels (e.g. `host_subject_id`) that would make the visualization cluttered (and are *probably* not of primary interest). 5 was chosen completely arbitrarily.

Do you have any ideas on how to best check the number of unique values per level? Maybe we add an optional parameter `min_count_per_level` under which values are discarded? i.e. if the minimum count was 3 and a level was only represented by 2 samples these samples would be dropped (with some explicit warning so users aren't confused when they have fewer samples/groups than expected)?

Comment (ahdilmore) [reply evident/interactive.py:42]: I like the idea of adding a `min_count_per_level` that behaves similarly to `max_levels_per_category`. I think that check could just be nested under this check, but let me know if you think of a better way to look for that?

**Comment by gibsramen** [comment]
@ahdilmore I added `min_count_per_level` and updated the README. Thoughts?

**Review by ahdilmore — APPROVED** [review]
Looks good to me 👍 Thanks Gibs!
